src/gt4py/next/iterator/ir_utils/ir_makers.py

```diff
@@ -401,3 +401,15 @@ def _impl(*its: itir.Expr) -> itir.FunCall:
 def map_(op):
     """Create a `map_` call."""
     return call(call("map_")(op))
+
+
+def as_fieldop(expr: itir.Expr, domain: Optional[itir.FunCall] = None) -> call:
+    """
+    Create an `as_fieldop` call.
+
+    Examples
+    --------
+    >>> str(as_fieldop(lambda_("it1", "it2")(plus(deref("it1"), deref("it2"))))("field1", "field2"))
+    '(⇑(λ(it1, it2) → ·it1 + ·it2))(field1, field2)'
+    """
+    return call(call("as_fieldop")(expr, *((domain,) if domain else ())))
```

Comment:
Maybe it'd be easier to read in this other way...  Feel free to ignore. 
```suggestion
    return call(call("as_fieldop")(*((expr, domain,) if domain else (expr,))))
```